Objective-C/Internal/CBLFleece.mm

```diff
@@ -63,8 +67,12 @@ static bool isOldAttachment(Dict properties) {
     static id createSpecialObjectOfType(Dict properties, DocContext *context) {
         slice type = properties.get(C4STR(kC4ObjectTypeProperty)).asString();
         if ((type && type == C4STR(kC4ObjectType_Blob)) || isOldAttachment(properties)) {
-            return [[CBLBlob alloc] initWithDatabase: context->database()
-                                          properties: context->toObject(properties)];
+            if (context->database())
+                return [[CBLBlob alloc] initWithDatabase: context->database()
+                                              properties: context->toObject(properties)];
+            else
+                // when setJSON has a blob, which lacks database. 
+                return [[CBLBlob alloc] initWithProperties: context->toObject(properties)];
```

Comment:
^ this will check for whether there is a database or not. if not, use the `initWithProperties` constructor